WooCommerce/WooCommerceTests/Mocks/MockCardReaderSettingsAlerts.swift

```diff
@@ -0,0 +1,47 @@
+import Foundation
+import UIKit
+@testable import WooCommerce
+
+enum MockCardReaderSettingsAlertsMode {
+    case cancelScanning
+    case closeScanFailure
+    case continueSearching
+    case connectFoundReader
+}
+
+final class MockCardReaderSettingsAlerts: CardReaderSettingsAlertsProvider {
+    private var mode: MockCardReaderSettingsAlertsMode
+
+    init(mode: MockCardReaderSettingsAlertsMode) {
+        self.mode = mode
+    }
+    func scanningForReader(from: UIViewController, cancel: @escaping () -> Void) {
+        if mode == .cancelScanning {
+            cancel()
+        }
+    }
+
+    func scanningFailed(from: UIViewController, error: Error, close: @escaping () -> Void) {
+        if mode == .closeScanFailure {
+            close()
+        }
+    }
+
+    func foundReader(from: UIViewController, name: String, connect: @escaping () -> Void, continueSearch: @escaping () -> Void) {
+        if mode == .continueSearching {
+            continueSearch()
+        }
+
+        if mode == .connectFoundReader {
+            connect()
+        }
+    }
+
+    func connectingToReader(from: UIViewController) {
+        // GNDN
```

Comment:
https://memory-alpha.fandom.com/wiki/Power_relay#:~:text=GNDN%20means%20'Goes%20nowhere%2C%20does,to%20include%20a%20GNDN%20pipe.